### PR TITLE
Remove v1alpha1

### LINKED ISSUE #1177: HNC: remove all trace of v1alpha1
Remove old types, conversions, and annotations (e.g. `subnamespaceOf`).

### PR DESCRIPTION
**Pull request by adrianludwin**
Remove the old API and all code related to the conversion. Searched for
all instances of "v1a1" and "v1alpha1" and removed them. Looked for
obsolete annotations like "managedBy" and "subnamespaceOf" and removed
those too.

Tested: all unit, integration and e2e tests pass

Fixes #1177

### CODE REVIEW

**Review by yiqigao217 — COMMENTED** [review]
lgtm just 1 question:

Do we want to make it back to `crd:triviaVersions` so there's no per-version schema, or you prefer leave it as is in case we have another api version in the future? See line `CRD_OPTIONS ?= "crd:trivialVersions=false"` in makefile.

**Comment by adrianludwin** [comment]
> lgtm just 1 question:
> 
> Do we want to make it back to `crd:triviaVersions` so there's no per-version schema, or you prefer leave it as is in case we have another api version in the future? See line `CRD_OPTIONS ?= "crd:trivialVersions=false"` in makefile.

We'll definitely have v1beta1 and then v1 one day so we should leave this as-is.

File: incubator/hnc/internal/validators/setup.go [hunk 1]
```diff
@@ -45,7 +43,6 @@ func CreateCertsIfNeeded(mgr ctrl.Manager, novalidation, internalCert bool) (cha
 		DNSName:        dnsName,
 		IsReady:        setupFinished,
 		VWHName:        vwhName,
-		CRDNames:       crds,
```
[the remark this answers is not in the document]
Comment (adrianludwin) [reply incubator/hnc/internal/validators/setup.go:48]: Yup! This is part of the base cert-controller library and AFAIK we were the only ones using this field.

**Pull request merged**